packages/bruno-app/src/providers/ReduxStore/slices/tabs.js

```diff
@@ -45,6 +45,34 @@ export const tabsSlice = createSlice({
     focusTab: (state, action) => {
       state.activeTabUid = action.payload.uid;
     },
+    switchTab: (state, action) => {
+      if (!state.tabs || !state.tabs.length) {
+        state.activeTabUid = null;
+        return;
+      }
+
+      const direction = action.payload.direction;
+
+      const activeTabIndex = state.tabs.findIndex((t) => t.uid === state.activeTabUid);
+
+      let toBeActivatedTabIndex = 0;
+
+      if (direction === 'pageup') {
+        if (activeTabIndex == 0) {
+          toBeActivatedTabIndex = state.tabs.length - 1;
+        } else {
+          toBeActivatedTabIndex = activeTabIndex - 1;
+        }
+      } else if (direction === 'pagedown') {
+        if (activeTabIndex == state.tabs.length - 1) {
+          toBeActivatedTabIndex = 0;
+        } else {
+          toBeActivatedTabIndex = activeTabIndex + 1;
+        }
+      }
+
+      state.activeTabUid = state.tabs[toBeActivatedTabIndex].uid;
```

Comment:
(dust = can be ignored) not my place to comment but I ended here after looking into ctrl-tab for Bruno, thanks for the implementation of the first shortcuts!

[I would recommend using the modulo operator here](https://www.joshwcomeau.com/javascript/modulo-operator/), that would simplify your code quite a bit  with taking care of the edge cases, something like:
```jsx
if (direction === 'pageup') {
  toBeActivatedTabIndex = (activeTabIndex - 1 + state.tabs.length) % state.tabs.length;
} else if (direction === 'pagedown') {
  toBeActivatedTabIndex = (activeTabIndex + 1) % state.tabs.length;
}

state.activeTabUid = state.tabs[toBeActivatedTabIndex].uid;

Reply:
Sure @Jeremie-Chauvel, I will update here soon.

Thanks for sharing the blogpost 😇, It is good.

Reply:
Hi @Jeremie-Chauvel, I pushed suggested modifications, you can review them. 😊